lib/event_store_client/connection.rb

```diff
@@ -79,5 +73,47 @@ def client
           host: host, port: port, per_page: per_page
         )
     end
+
+    def read_from_stream(stream, direction:, start:, count:, resolve_links:)
+      response =
+        client.read(
+          stream, start: start, direction: direction, count: count, resolve_links: resolve_links
+        )
+      return [] if response.body.nil? || response.body.empty?
+      JSON.parse(response.body)['entries'].map do |entry|
+        deserialize_event(entry)
+      end.reverse
+    end
+
+    def read_all_from_stream(stream, resolve_links:)
+      count = 20
```

Comment:
I would move to the const like DEFAULT_COUNT = 20 DEFAULT_START_POSITION = 0, and probably would be good to pass start position as a parameter because maybe we will want to read the whole stream starting at a specific position, but it's only my opinion.

Reply:
Well, this is an adapter to the gregg's event store. We should map the functionality to what the event store API actually allows to do, even if we won't use it at the moment. So yes, I agree, that we should add the starting position parameter.